.github/workflows/upgrade_protbuf_release.yml

```diff
@@ -0,0 +1,106 @@
+# 
+# pipeline to check for new release of Protocol Buffers and update this repository accordingly
+#
+name: Import External Release of Protocol Buffers
+
+on:
+  workflow_dispatch: 
+  schedule:
+    - cron: '30 0 1 * *'
+jobs:
+  import_release:
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout Repository
+          uses: actions/checkout@v2
+
+        - name: "Check for New Release"
+          id: check_release
+          run: |
+            latest_release=$(curl -s https://api.github.com/repos/protocolbuffers/protobuf/releases/latest | jq -r '.tag_name')
+            current_protobuf_rel=$(cat 02_mqtt-cluster/protobuf/protobuf.version)
+            if [[ "$current_protobuf_rel" != "$latest_release" ]]; then
+              echo "new_release=true"  >> $GITHUB_OUTPUT
+              echo "latest_release=${latest_release}"  >> $GITHUB_OUTPUT
+              echo "current_protobuf_rel=${current_protobuf_rel}"  >> $GITHUB_OUTPUT
+            else
+              echo "new_release=false"  >> $GITHUB_OUTPUT
+            fi
+        - name: Create New Branch
+          id: create_branch
+          if: steps.check_release.outputs.new_release == 'true'
+          env:
+            latest_release: ${{ steps.check_release.outputs.latest_release }}
+
+          run: |
+            git config --global user.email "actions@github.com"
+            git config --global user.name "Github Actions"
+            BRANCH_NAME="features/upgrade_protobuf_to_${{env.latest_release}}"
+            git checkout -b $BRANCH_NAME
+            
+            git push origin $BRANCH_NAME
+            echo "branch_name=${BRANCH_NAME}" >> $GITHUB_OUTPUT
+
+        - name: Download and Extract Release
+          if: steps.check_release.outputs.new_release == 'true'
+          env:
+            latest_release: ${{ steps.check_release.outputs.latest_release }}
+            current_protobuf_rel: ${{ steps.check_release.outputs.current_protobuf_rel }}
+            BRANCH_NAME:  ${{ steps.create_branch.outputs.BRANCH_NAME }}
+          run: |
+            git config --global user.email "actions@github.com"
+            git config --global user.name "Github Actions"
+
+            cd 02_mqtt-cluster
+            mkdir temp
+            cd temp
+            OLDIFS=$IFS; 
+            IFS=','; 
+            # Download 3 executables, rename them with proper suffix 
+            for i in linux-x86_64,"" win64,"" osx-aarch_64,_osx; 
```

Comment:
@mkashwin may be move them to array and user a variable rather than using ${1} or ${2}

Reply:
Done